app/controllers/api/v1/users_controller.rb

```diff
@@ -28,10 +28,13 @@ def destroy
   private
 
   def user_params
-    # needs to be cleaned up
-    a = params.permit(:uid, :provider)
-    b = params.require(:credentials).permit(:token, :refresh_token)
-    c = params.require(:info).permit(:email)
-    a.merge(b).merge(c)
+    if params[:action] = 'update'
+      params.permit(:id, :email)
+    else
+      a = params.permit(:uid, :provider)
+      b = params.require(:credentials).permit(:token, :refresh_token)
+      c = params.require(:info).permit(:email)
+      a.merge(b).merge(c)
+    end
```

Comment:
Update action would not work due to :credentials and :info not being passed. Those are not wanting to be updated when user updates their profile